core/crypto/keymaster/signer.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keymaster
+
+import (
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/google/keytransparency/core/crypto/signatures"
+	"github.com/google/keytransparency/core/crypto/signatures/factory"
+	kmpb "github.com/google/keytransparency/core/proto/keymaster"
+)
+
+type signer struct {
+	signatures.Signer
+	addedAt     time.Time // time when key is added to keymaster.
+	description string
+	status      kmpb.SigningKey_KeyStatus
+}
+
+// NewSigner creates a signer object from a private key.
+func NewSigner(s signatures.Signer, addedAt time.Time,
+	description string, status kmpb.SigningKey_KeyStatus) Signer {
+	return &signer{
+		Signer:      s,
+		addedAt:     addedAt,
+		description: description,
+		status:      status,
+	}
+}
+
+// NewSignerFromPEM parses a PEM formatted block and returns a signer object created
+// using that block.
+func NewSignerFromPEM(pemKey []byte) (Signer, error) {
+	s, err := factory.NewSignerFromPEM(pemKey)
+	if err != nil {
+		return nil, err
+	}
+	return &signer{
+		Signer:      s,
+		addedAt:     time.Now(),
+		description: "Signer created from PEM",
+		status:      kmpb.SigningKey_ACTIVE,
+	}, nil
+}
+
+// SignerFromRawKey creates a signer object from given raw key bytes.
+func SignerFromRawKey(b []byte) (signatures.Signer, error) {
```

Comment:
Please consider starting the name of all functions that create `keymaster.Signer`, `keymaster.Verifier`, `signatures.Signer`, and `signatures.Verifier` objects with `New`, e.g., `NewSignerFromPEM`, `NewVerifierFromRawKey`, etc. We currently have some of these functions start with `New` and others don't. I might have introduced this inconsistency when I worked on the `signatures` library a while ago and now I can see that it might cause some confusion in the future.